raster/r.external/proj.c

```diff
@@ -57,18 +104,16 @@ void check_projection(struct Cell_head *cellhd, GDALDatasetH hDS,
 		authkey = "GEOGCS";
 
 	    authname = OSRGetAuthorityName(hSRS, authkey);
-	    if (authname && *authname && strcmp(authname, "EPSG") == 0) {
+	    if (authname && *authname) {
 		authcode = OSRGetAuthorityCode(hSRS, authkey);
 		if (authcode && *authcode) {
-		    G_debug(1, "found EPSG:%s", authcode);
-		    proj_epsg = G_create_key_value();
-		    G_set_key_value("epsg", authcode, proj_epsg);
-		}
+		    G_asprintf(&srid, "%s:%s", authname, authcode);
 	    }
 	}
 	if (hSRS)
 	    OSRDestroySpatialReference(hSRS);
     }
+#endif
     else {
```

Comment:
```suggestion
    }
    else {
```

Reply:
`#endif` should be placed after `else` block

Reply:
No, the `#endif` must be before the `else` block. Above, there are two different `if` blocks, one for GDAL >= 3, another one for GDAL < 3.